### PR TITLE
Optional JSON parse

### LINKED ISSUE #241: JSON Parsing should be optional
Parsing JSON as part of the consumer should not be part of the responsibility of the consumer. First of all, it tries to parse JSON if it looks like it might maybe possibly have JSON. If that rough assumption is wrong, it throws a parser error which is not expected from a user of the library perspective. There are multiple errors that can occur (syntax errors, out of memory) that a consumer user might want to deal with specifically. The consumer user might also want to use more optimized JSON parsers if they know the schema.

I would like to suggest that either the consumer code should NOT parse the JSON automatically or that it should be optional.

https://github.com/lifion/lifion-kinesis/blob/0ae854fb335275fcdec93746453dd3cd480110c9/lib/records.js#L85

### PR DESCRIPTION
**Pull request by jennyEckstein**
Add optional JSON parse.

Fixes #241

### CODE REVIEW

File: lib/records.js [hunk 1]
```diff
@@ -191,11 +203,12 @@ class RecordsDecoder extends Transform {
    *
    * @param {object} options - The initialization options.
    * @param {string} options.compression - The kind of compression to use in records data.
+   * @param {string} options.shouldParseJson - If data is in JSON format and should be parsed.
    * @param {string} options.logger - An instance of a logger.
    * @param {object} options.s3Client - The s3Client in the current kinesis client.
    * @param {boolean} options.useS3ForLargeItems - Whether to automatically use an S3 bucket to store large items or not.
    */
-  constructor({ compression, logger, s3Client, useS3ForLargeItems }) {
+  constructor({ compression, logger, s3Client, shouldParseJson = 'auto', useS3ForLargeItems }) {
```
Comment (eaviles) [inline lib/records.js]: This is already defaulted somewhere else.

File: lib/index.js [hunk 2]
```diff
@@ -230,6 +230,8 @@ class Kinesis extends PassThrough {
    *        S3 bucket is tagged with these tags. If the bucket already has tags, they will be merged.
    * @param {number} [options.shardCount=1] - The number of shards that the newly-created stream
    *        will use (if the `createStreamIfNeeded` option is set)
+   * @param {string|boolean} [options.shouldParseJson=auto] - Indicates if consumer data should be parsed as JSON.
+   *        'auto': if data matches json format, data will be parsed. `true`: will parse data regardless
```
Comment (eaviles) [inline lib/index.js]: Maybe:

> Whether if retrieved records' data should be parsed as JSON or not. Set to "auto" to only attempt parsing if data looks like JSON.

File: lib/records.js [hunk 3]
```diff
@@ -81,7 +89,11 @@ function getRecordsDecoder({ compression, inputEncoding, logger, s3Client, useS3
         data = Buffer.from(data, 'base64').toString('utf8');
       }
 
-      if (IS_JSON_REGEX.test(data)) {
+      if (shouldParseJson === 'auto') {
+        if (IS_JSON_REGEX.test(data)) {
+          data = JSON.parse(data);
+        }
+      } else if (shouldParseJson === true) {
```
Comment (eaviles) [inline lib/records.js]: What about:

```js
if (shouldParseJson === 'auto' && IS_JSON_REGEX.test(data) || shouldParseJson === true) {
```

File: lib/records.js (hunk 1 above)
Comment (jennyEckstein) [reply lib/records.js]: fixed

File: lib/index.js (hunk 2 above)
Comment (jennyEckstein) [reply lib/index.js]: fixed

File: lib/records.js (hunk 3 above)
Comment (jennyEckstein) [reply lib/records.js]: fixed

**Pull request merged**